core/server/common/src/main/java/alluxio/cli/validation/UserLimitValidationTask.java

```diff
@@ -0,0 +1,101 @@
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+/**
+ * Task for validating system limit for current user.
+ */
+public final class UserLimitValidationTask implements ValidationTask {
+  private static final int NUMBER_OF_OPEN_FILES_MIN = 16384;
+  private static final int NUMBER_OF_OPEN_FILES_MAX = 800000;
+  private static final int NUMBER_OF_USER_PROCESSES_MIN = 16384;
+  private final String mCommand;
+  private final String mName;
+  private final Integer mLowerBound;
+  private final Integer mUpperBound;
+
+  private UserLimitValidationTask(String name, String command,
+                                  Integer lowerBound, Integer upperBound) {
+    mName = name;
+    mCommand = command;
+    mLowerBound = lowerBound;
+    mUpperBound = upperBound;
+  }
+
+  @Override
+  public boolean validate() {
+    try {
+      // checks if the current user is root
+      Process process = Runtime.getRuntime().exec(new String[] {"bash", "-c", mCommand});
+      try (BufferedReader processOutputReader = new BufferedReader(
+          new InputStreamReader(process.getInputStream()))) {
+        String line = processOutputReader.readLine();
+        if (line == null) {
+          System.err.format("Unable to check user limit for %s.%n", mName);
+          return false;
+        }
+
+        if (line.equals("unlimited")) {
+          if (mUpperBound != null) {
+            System.err.format("The user limit for %s is unlimited. It should be less than %d%n",
+                mName, mUpperBound);
+            return false;
+          }
+
+          return true;
+        }
+
+        int value = Integer.parseInt(line);
+        if (mUpperBound != null && value > mUpperBound) {
+          System.err.format("The user limit for %s is too large. The current value is %d. "
+              + "It should be less than %d%n", mName, value, mUpperBound);
+          return false;
+        }
+
+        if (mLowerBound != null && value < mLowerBound) {
+          System.err.format("The user limit for %s is too small. The current value is %d. "
+              + "It should be bigger than %d%n", mName, value, mLowerBound);
+          return false;
+        }
+
+        return true;
+      }
+    } catch (IOException e) {
+      System.err.format("Unable to check user limit for %s: %s.%n", mName, e.getMessage());
+      return false;
+    }
+  }
+
+  /**
+   * Creates a validation task for checking whether the user limit for number of open files
+   * is within reasonable range.
+   * @return the validation task for this check
+   */
+  public static ValidationTask createNumberOfOpenFilesLimitValidationTask() {
+    return new UserLimitValidationTask("number of open files", "ulimit -n",
+        NUMBER_OF_OPEN_FILES_MIN, NUMBER_OF_OPEN_FILES_MAX);
```

Comment:
It is generally recommended against setting a very large file descriptor limit. User would rarely have that many open files under normal usage. If they ever use that many, it would likely to be an issue leaking file descriptors. Giving a large limit will mask the real issue and cause problems elsewhere.